src/mpi/coll/bcast/bcast_intra_scatter_recursive_doubling_allgather.c

```diff
@@ -98,7 +92,8 @@ int MPIR_Bcast_intra_scatter_recursive_doubling_allgather(void *buffer,
     }
 
     /* curr_size is the amount of data that this process now has stored in
-     * buffer at byte offset (relative_rank*scatter_size) */
+     * buffer at byte offset (relative_rank*scatter_size)
+     * This value can go -ve for higher ranks */
```

Comment:
```suggestion
    /* Note: when nbytes < comm_size, higher ranks will have 0 data and nbytes - offset may become negative */
```

Reply:
When nbytes=8 and comm_size=6,
we are in rank = relative_rank = 5, then  scatter_size  will be 2. we get -2 i.e 8-5*2.
That is why I mentioned it as higher values for rank.

Reply:
I see. In your example, we are distributing the data `2, 2, 2, 2, 0, 0` rather than what I thought should be `2, 2, 1, 1, 1, 1`.  How about this comment:
```
/* Note: since we are rounding up scatter_size, higher ranks may not have data and nbytes-offset may be negative */
```
?

Reply:
That looks good. Let me update with that comment. Thank you.